pubsub_logging/async_handler.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Python logging handler implementation for Cloud Pub/Sub.
+
+This module provides a logging.Handler implementation which sends the
+logs to Cloud Pub/Sub[1] asynchronously. The logs are kept in an
+internal queue and child workers will pick them up and send them in
+background.
+
+[1]: https://cloud.google.com/pubsub/docs
+
+"""
+
+import logging
+import multiprocessing as mp
+
+# For Python 2 and Python 3 compatibility.
+try:
+    from queue import Empty
+except ImportError:
+    from Queue import Empty
+
+from pubsub_logging import errors
+
+from pubsub_logging.utils import compat_urlsafe_b64encode
+from pubsub_logging.utils import get_pubsub_client
+from pubsub_logging.utils import publish_body
+
+
+BATCH_SIZE = 1000
+DEFAULT_WORKER_SIZE = 1
+DEFAULT_RETRY_COUNT = 10
+DEFAULT_TIMEOUT = 1
+
+
+class AsyncPubsubHandler(logging.Handler):
```

Comment:
This will definitely make the code simpler, but what I'm afraid is that, Pool class starts multiple threads which are not necessary for my use case. I would rather only use the minimum mp primitives necessary for my use case.

WDYT?